beta-src/src/components/ui/WDPress.tsx

```diff
@@ -1,34 +1,176 @@
 import * as React from "react";
-import { Box } from "@mui/material";
+import {
+  Box,
+  Stack,
+  IconButton,
+  TextField,
+  ButtonGroup,
+  Divider,
+} from "@mui/material";
+import { Email, Send } from "@mui/icons-material";
+
+import Button from "@mui/material/Button";
 import Device from "../../enums/Device";
 import useViewport from "../../hooks/useViewport";
 import getDevice from "../../utils/getDevice";
+import useInterval from "../../utils/useInterval";
+import WDMessageList from "./WDMessageList";
+import { CountryTableData } from "../../interfaces";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import {
+  fetchGameMessages,
+  gameApiSliceActions,
+  gameMessages,
+  gameOverview,
+  markMessagesSeen,
+  sendMessage,
+} from "../../state/game/game-api-slice";
 
 interface WDPressProps {
   children: React.ReactNode;
+  userCountry: CountryTableData;
+  countries: CountryTableData[];
 }
 
 const WDPress: React.FC<WDPressProps> = function ({
   children,
+  userCountry,
+  countries,
 }): React.ReactElement {
   const [viewport] = useViewport();
   const device = getDevice(viewport);
+  const dispatch = useAppDispatch();
   const mobileLandscapeLayout =
     device === Device.MOBILE_LANDSCAPE ||
     device === Device.MOBILE_LG_LANDSCAPE ||
     device === Device.MOBILE;
   const padding = mobileLandscapeLayout ? "0 6px" : "0 16px";
-  const width = mobileLandscapeLayout ? 272 : 358;
+
+  const [userMsg, setUserMsg] = React.useState("");
+  const [countryIDSelected, setCountryIDSelected] = React.useState(
+    // start with the first country
+    Math.min(...countries.map((country) => country.countryID)),
+  );
+
+  const { user, gameID } = useAppSelector(gameOverview);
+  const messages = useAppSelector(gameMessages);
+
+  // FIXME: for now, crazily fetch all messages every 1sec
+  useInterval(() => {
+    if (user && gameID && messages && messages.outstandingRequests === 0) {
+      dispatch(gameApiSliceActions.updateOutstandingMessageRequests(1));
+      dispatch(
+        fetchGameMessages({
+          gameID: gameID as unknown as string,
+          countryID: user.member.countryID as unknown as string,
+          allMessages: "true",
+          sinceTime: messages.time as unknown as string,
+        }),
+      );
+    }
+  }, 1000);
```

Comment:
@s0mya @adamlerer I have to agree with Somy's concern here, not sure how well this well play out pushing it to the test server. Generally the requests seem to take slightly longer in the test server than they do on local, though it was expressed in Kestas' email that their servers are underwhelmed. 

Would you be able to hook this up to the existing polling? The existing polling makes API calls based on various criteria already: 
1. How many users are online, example if 5/7 players are currently logged in it will make more calls
2. How many users are ready, example if 2/7 players currently readied their orders
3. How long the game is, example quicker games like 5 or 10 minutes will have increased polling frequency
4. User activity, example if the user is not active (afk) then it will stop polling

Let me know if you have any questions about this or want me to clarify anything.

Reply:
I'm also worried about load on the server, but I felt that in the short term it's best to try it out and see if it's a problem. Notice that I implemented a server-side fast path through memcache so it should be relatively cheap if you have no messages.

I don't think it's enough to look at `newMessages`, because how will you know when a new message just came in?

The first use case for this UI is a 5-minute-per-phase tournament, so the messaging must be very responsive (since you're going to send/receive lots of messages in each 5 minute window).

One good suggestion Alex had is that if the polling is too expensive, as a fallback we could do less frequent polling and have a manual (rate-limited to 1/s) Refresh button in the chat window.